### PR TITLE
Keepers [Don't Merge]

### PR DESCRIPTION
**Pull request by Petethegoat**
Keepers second draft. Still plenty left to do. Wildly untested and unfinished. Lots of quick code.
Features right now are:
- Two hands that work as human hands do (or monkey hands, to be more accurate. They can't use computers.)
- Binary speech only.
- Vent crawling.
- All access.
- Basic hud.
- Laws, yo
- Keeper cannibalism.

Still need to add spawning and emagging. Maybe more stuff with speech. Maybe tools and shit.

Minor refactoring of proc/swap_hand(), definition is moved to /mob.
Adjustments to hotkeys.
Added var/access_card to simple_animal, they can now utilize access.

### CODE REVIEW

File: code/modules/mob/living/simple_animal/friendly/keeper.dm [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+		if(hand)	//This being 1 means the left hand is in use
+			hud_used.l_hand_hud_object.icon_state = "hand_l_active"
+			hud_used.r_hand_hud_object.icon_state = "hand_r_inactive"
+		else
+			hud_used.l_hand_hud_object.icon_state = "hand_l_inactive"
+			hud_used.r_hand_hud_object.icon_state = "hand_r_active"
+	return
+
+
+/mob/living/simple_animal/keeper/put_in_l_hand(obj/item/I)
```
Comment (Petethegoat) [inline code/modules/mob/living/simple_animal/friendly/keeper.dm:89]: these procs will use oop soon i'm aware that this is awful

**Comment by PinkyGatchabee** [comment]
kawaii

**Comment by RemieRichards** [comment]
@dannno Well pete is a brilliant coder.

File: code/modules/mob/living/simple_animal/friendly/keeper.dm [hunk 2]
```diff
@@ -0,0 +1,111 @@
+/mob/living/simple_animal/keeper
+	name = "keeper"
+	desc = "A keeper droid. They're kawaii as fuck and they'll repair up your shit."
```
Comment (Giacom) [inline code/modules/mob/living/simple_animal/friendly/keeper.dm]: muh immersion, could you make a sensible description?

Comment (Petethegoat) [inline code/modules/mob/living/simple_animal/friendly/keeper.dm]: NO

Comment (Iamgoofball) [inline code/modules/mob/living/simple_animal/friendly/keeper.dm]: plz dont ever use the word kawaii again

kthx

also yeah as great that description is i'm with giacom on this one. switch the description.

Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/friendly/keeper.dm]: @Iamgoofball Atashi wa cho kawaii, anata wa baka desu!

**Comment by Petethegoat** [comment]
Updoot. Ready for testing now. There's probably a lot of shit that won't work, so the more we find the better.

**Comment by Atskadan** [comment]
wow look at that tiny bucket
that tiny extinguisher
that tiny little dude
hey dude whats up how are you 
gonna go fight some fires hahahah ok little guy have fun

File: code/modules/mob/mob_movement.dm [hunk 3]
```diff
@@ -31,15 +31,10 @@
...
-		var/mob/living/carbon/C = usr
-		if(!C.get_active_hand())
-			usr << "\red You have nothing to drop in your hand."
-			return
-		drop_item()
-	else
-		usr << "\red This mob type cannot drop items."
-	return
+	if(!usr.get_active_hand())
+		usr << "\red You have nothing to drop in your hand."
```
Comment (Atskadan) [inline code/modules/mob/mob_movement.dm:35]: SPAN CLASSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS

Comment (Petethegoat) [inline code/modules/mob/mob_movement.dm:35]: aa AAAAAAAA

**Comment by Iamgoofball** [comment]
can't automerge

did we touch the map at all in other PRs? as the food one barfed too

**Comment by Atskadan** [comment]
it doesnt matter if you cant automerge because it says [DONT MERGE] in the title
its just because he hasnt pulled the most recent changes to his repo

**Comment by Neerti** [comment]
I hate the sprite personally but that's no reason to deny something.
:+1:

**Comment by JJRcop** [comment]
pls hury and feenish

**Comment by Petethegoat** [comment]
yur hur bluh

soon maybe

**Comment by Atskadan** [comment]
> pls hury and feenish

**Comment by ghost** [comment]
Please don't give them all access, they can vent crawl to get around plenty fine. There's no reason they'll need to use doors outside of maint doors in emergencies. All access will lead to grief and vent crawling is already all access.

**Comment by Iamgoofball** [comment]
Can't ventcrawl if a pipe grts fucked up I think.

Plus if you're building a room and put a door in but no vents you're never able to get back in bcause no talking to other entities besides other mommis

The all access ensures 100% automacy

**Comment by Iamgoofball** [comment]
Plus you'd alsi need to remove ventcrawling if you're going by the grief excuse

**Comment by vista-** [comment]
@Petethegoat, any updates on this?

**Comment by RemieRichards** [comment]
Bay are beating us https://github.com/Baystation12/Baystation12/pull/4941

**Comment by RemieRichards** [comment]
If There is or if Pete can chalk me up some kind of "This isn't done but is wanted" Eg, a checklist or something.

But Yeah I wouldn't mind taking it on.

**Comment by RemieRichards** [comment]
Pete's current code is Nice. This should be fun and easy to finish.

**Comment by GunHog** [comment]
Is this still alive, or what?

**Comment by Atskadan** [comment]
pete is 2whipped2code

**Comment by RemieRichards** [comment]
Pete isn't ass, pete just needs a rest and a long holiday I think.
and Yeah, I probably will pick it up.

**Comment by ghost** [comment]
looking forward to it!

**Comment by ghost** [comment]
I am no longer looking forward to it.

Pete is now startoad-tier I guess.

**Comment by Atskadan** [comment]
lol
no, he just quit

**Comment by ghost** [comment]
But he also said he'd finish this. I guess we can wait till RR does it, but RR's in the middle of some other shit right now. I can w8 if it means some quality code behind this.

**Comment by GunHog** [comment]
We reeeeeeally need someone else to take this up. :(

**Comment by ghost** [comment]
Meh. RR finished his mutations thing, and is doing xenoarch right now. Again, I can wait.

**Comment by RemieRichards** [comment]
> his

:/

**Comment by ghost** [comment]
:\

**Comment by GunHog** [comment]
Someone finish plz. That or loot Bay's version!

**Comment by vista-** [comment]
_cough_ /vg/ _cough_ better _cough_ fun _cough_ theymadetheoriginalanyway
_cough_

It's the flu.

--Zsolt Hegyi

**Comment by GunHog** [comment]
Or /vg/'s version! Just please get something moving! I feel as if this PR is never going to get finished, then closed because it was left to rot. It should not go the way of Dismemberment!

**Comment by vista-** [comment]
The current code is not compatible with /VG/'s as they used a silicon-based
mob instead of a simple_animal. A full, new port would be required. If said
port begins I suggest that you close the PR.

--Zsolt Hegyi

**Comment by ghost** [comment]
Gun hog do it yourself seriously if you have such a huge boner for mommis

**Comment by RemieRichards** [comment]
If anyone DOES attempt to port /vg/'s MoMMIs, I tried it the other day, and parts of the code are redundant (or in the case of 2 definitions, flat out wrong) and there's lots of copy pasted code.
and quite a few of the pieces of code present in /vg/'s MoMMIs is incompatible so, Enjoy. If anything I'd say the best bet is to finish Pete's code instead, it's more compatible.

**Comment by ike709** [comment]
So is this abandoned now or what?

**Comment by RemieRichards** [comment]
I wonder why I'm closing this PR? hehe.

**Comment by Anton-Kr** [comment]
RIP NTStation.

**Pull request closed without merging**